geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/maps/package.scala

```diff
@@ -0,0 +1,12 @@
+package org.locationtech.geomesa.utils
+
+package object maps {
+
+  def aggregateMaps[K,V](maps: TraversableOnce[collection.Map[K,V]], operation: (V,V) => V, emptyVal: V): Map[K,V] =
+    maps.flatMap(_.toSeq).foldLeft(Map[K,V]()){ case (mapSoFar, (k, v)) =>
+      mapSoFar.updated(k, operation(mapSoFar.getOrElse(k, emptyVal), v))
+    }
```

Comment:
My general impression/understanding of immutable data structures in scala is that when you create a "new" immutable map, etc, by copying an old one, the new one internally reuses most of the data structures from the old one, only changing the parts of the map where data is different.  That being said, I don't have a good feel for how to tell when it is actually more efficient to use a mutable map, so I'll go ahead and change it here and above.